geode-core/src/main/java/org/apache/geode/management/internal/configuration/converters/RegionConverter.java

```diff
@@ -68,27 +61,90 @@ protected RegionConfig fromNonNullConfigObject(Region configObject) {
     return region;
   }
 
+  /**
+   * Data policy to regionType is almost a 1-to-1 mapping, except in
+   * the case of DataPolicy.PARTITION, we will need to see the local max memory
+   * to determine if it's a PARTITION type or a PARTITION_PROXY type.
+   *
+   * we do our best to infer the type from the existing xml attributes. For data
+   * policies not supported by management rest api (for example, NORMAL and RELOADED)
+   * it will show as UNSUPPORTED
+   */
+  public RegionType getRegionType(RegionAttributesType regionAttributes) {
+    if (regionAttributes == null) {
+      return RegionType.UNSUPPORTED;
+    }
+    RegionAttributesDataPolicy dataPolicy = regionAttributes.getDataPolicy();
+
+    if (regionAttributes.getDataPolicy() == null) {
+      return RegionType.UNSUPPORTED;
+    }
+
+    switch (dataPolicy.name()) {
+      case "PARTITION": {
+        RegionAttributesType.PartitionAttributes partitionAttributes =
+            regionAttributes.getPartitionAttributes();
+        if (partitionAttributes != null && partitionAttributes.getLocalMaxMemory().equals("0")) {
```

Comment:
Should you also handle a null and empty string returned by getLocalMaxMemory() here?
I think both of these would indicate a default value and the default for this one should be zero.